gateway-service/src/main/java/org/zowe/apiml/gateway/filters/TomcatFilter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.filters;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.*;
+import jakarta.servlet.annotation.WebFilter;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.message.core.Message;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;
+
+import java.io.IOException;
+import java.net.URLDecoder;
+
+/**
+ * This filter checks if encoded slashes in the URI are allowed based on configuration.
+ * If not allowed and encoded slashes are present, it returns a BAD_REQUEST response.
+ */
+@Component
+@WebFilter
+@RequiredArgsConstructor
+public class TomcatFilter implements Filter {
```

Comment:
Maybe better to rewrite it using WebFilter?

Reply:
good point, done